Mathlib/RingTheory/AlgebraicIndependent.lean

```diff
@@ -566,6 +704,40 @@ theorem IsTranscendenceBasis.isAlgebraic [Nontrivial R] (hx : IsTranscendenceBas
   exact h₂ (hx.2 (Set.range fun o : Option ι => o.elim a x)
     ((algebraicIndependent_subtype_range ai.injective).2 ai) h₁)
 
+/-- If `x` is a transcendence basis of `A/R`, then it is empty if and only if
+`A/R` is algebraic. -/
+theorem IsTranscendenceBasis.isEmpty_iff_isAlgebraic [Nontrivial R]
+    (hx : IsTranscendenceBasis R x) :
+    IsEmpty ι ↔ Algebra.IsAlgebraic R A := by
+  refine ⟨fun _ ↦ ?_, fun _ ↦ hx.1.isEmpty_of_isAlgebraic⟩
+  have := hx.isAlgebraic
+  rw [Set.range_eq_empty x, adjoin_empty] at this
+  refine Algebra.IsAlgebraic.of_ringHom_of_comp_eq (algebraMap R (⊥ : Subalgebra R A))
+    (RingHom.id A) (fun r ↦ ?_) Function.injective_id (by ext; rfl)
+  obtain ⟨y, hy⟩ := Algebra.mem_bot.1 r.2
+  use y
+  apply_fun algebraMap _ A using Subtype.val_injective
+  rw [← IsScalarTower.algebraMap_apply, hy]; rfl
```

Comment:
```suggestion
    (RingHom.id A) ?_ Function.injective_id (by ext; rfl)
  rintro ⟨_, r, rfl⟩; exact ⟨r, rfl⟩
```
and I think we should extract the lemma `Algebra.IsAlgebraic (↥⊥) A ↔ Algebra.IsAlgebraic R A`.

Reply:
Done. The results are extracted into 'Mathlib/RingTheory/Algebraic.lean`. Note that some of the corresponding results for `IntermediateField` are not added yet; they should probably go to `FieldTheory/Adjoin.lean`.